src/Acquisition/_Acquisition.c

```diff
@@ -1120,96 +1167,121 @@ static PyMappingMethods Wrapper_as_mapping = {
 
 /* Code to access Wrapper objects as numbers */
 
-static PyObject *
-Wrapper_sub(Wrapper *self, PyObject *o)
-{
-  return CallMethodO(OBJECT(self),py__sub__,Build("(O)", o),NULL);
-}
+#define WRAP_UNARYOP(OPNAME) \
+    static PyObject* Wrapper_##OPNAME(PyObject* self) { \
+        return PyObject_CallMethod(self, "__"#OPNAME"__", NULL); \
+    }
 
-static PyObject *
-Wrapper_div(Wrapper *self, PyObject *o)
-{
-  return CallMethodO(OBJECT(self),py__div__,Build("(O)", o),NULL);
-}
+#define WRAP_BINOP(OPNAME) \
+    static PyObject* Wrapper_##OPNAME(PyObject* self, PyObject* o1) { \
+        return PyObject_CallMethod(self, "__"#OPNAME"__", "O", o1); \
+    }
 
-static PyObject *
-Wrapper_mod(Wrapper *self, PyObject *o)
-{
-  return CallMethodO(OBJECT(self),py__mod__,Build("(O)", o),NULL);
-}
+#define WRAP_TERNARYOP(OPNAME) \
+    static PyObject* Wrapper_##OPNAME(PyObject* self, PyObject* o1, PyObject* o2) { \
+        return PyObject_CallMethod(self, "__"#OPNAME"__", "OO", o1, o2); \
+    }
 
-static PyObject *
-Wrapper_divmod(Wrapper *self, PyObject *o)
-{
-  return CallMethodO(OBJECT(self),py__divmod__,Build("(O)", o),NULL);
-}
+WRAP_BINOP(sub);
+WRAP_BINOP(mul);
 
-static PyObject *
-Wrapper_pow(Wrapper *self, PyObject *o, PyObject *m)
-{
-  return CallMethodO(OBJECT(self),py__pow__,Build("(OO)", o, m),NULL);
-}
+#ifndef PY3K
+WRAP_BINOP(div);
+#endif
 
-static PyObject *
-Wrapper_neg(Wrapper *self)
-{
-  return CallMethodO(OBJECT(self), py__neg__, NULL, NULL);
-}
+WRAP_BINOP(mod);
+WRAP_BINOP(divmod);
+WRAP_TERNARYOP(pow);
+WRAP_UNARYOP(neg);
+WRAP_UNARYOP(pos);
+WRAP_UNARYOP(abs);
+WRAP_UNARYOP(invert);
+WRAP_BINOP(lshift);
+WRAP_BINOP(rshift);
+WRAP_BINOP(and);
+WRAP_BINOP(xor);
+WRAP_BINOP(or);
+
+WRAP_UNARYOP(int);
+
+#ifndef PY3K
+WRAP_UNARYOP(long);
+#endif
 
-static PyObject *
-Wrapper_pos(Wrapper *self)
-{
-  return CallMethodO(OBJECT(self), py__pos__, NULL, NULL);
-}
+WRAP_UNARYOP(float);
 
-static PyObject *
-Wrapper_abs(Wrapper *self)
-{
-  return CallMethodO(OBJECT(self), py__abs__, NULL, NULL);
-}
+#ifndef PY3K
+WRAP_UNARYOP(oct);
+WRAP_UNARYOP(hex);
+#endif
 
-static PyObject *
-Wrapper_invert(Wrapper *self)
-{
-  return CallMethodO(OBJECT(self), py__invert__, NULL, NULL);
-}
+WRAP_BINOP(iadd);
+WRAP_BINOP(isub);
+WRAP_BINOP(imul);
 
-static PyObject *
-Wrapper_lshift(Wrapper *self, PyObject *o)
-{
-  return CallMethodO(OBJECT(self),py__lshift__,Build("(O)", o),NULL);
-}
+#ifndef PY3K
+WRAP_BINOP(idiv);
+#endif
 
-static PyObject *
-Wrapper_rshift(Wrapper *self, PyObject *o)
-{
-  return CallMethodO(OBJECT(self),py__rshift__,Build("(O)", o),NULL);
-}
+WRAP_BINOP(imod);
+WRAP_TERNARYOP(ipow);
+WRAP_BINOP(ilshift);
+WRAP_BINOP(irshift);
+WRAP_BINOP(iand);
+WRAP_BINOP(ixor);
+WRAP_BINOP(ior);
+WRAP_BINOP(floordiv);
+WRAP_BINOP(truediv);
+WRAP_BINOP(ifloordiv);
+WRAP_BINOP(itruediv);
+WRAP_UNARYOP(index);
+WRAP_BINOP(matmul);
+WRAP_BINOP(imatmul);
```

Comment:
Could we make the i/matmul wrapper definitions dependent on the Python version via ``#if ((PY_MAJOR_VERSION == 3) && (PY_MINOR_VERSION > 4))`` as well? So far only adding them to the `PyNumberMethods` checks the version.

My compiler complains with unused function warnings:
```
src/Acquisition/_Acquisition.c:1238:1: warning: unused function 'Wrapper_matmul' [-Wunused-function]
WRAP_BINOP(matmul);
^
src/Acquisition/_Acquisition.c:1176:22: note: expanded from macro 'WRAP_BINOP'
    static PyObject* Wrapper_##OPNAME(PyObject* self, PyObject* o1) { \
                     ^
<scratch space>:94:1: note: expanded from here
Wrapper_matmul
^
```